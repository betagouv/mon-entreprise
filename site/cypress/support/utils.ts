export const fr = Cypress.env('language') === 'fr'
export const baseUrl = Cypress.config('baseUrl')
