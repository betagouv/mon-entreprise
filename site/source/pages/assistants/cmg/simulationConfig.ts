import { SimulationConfig } from '@/domaine/SimulationConfig'

export const configCMG: SimulationConfig = {}
