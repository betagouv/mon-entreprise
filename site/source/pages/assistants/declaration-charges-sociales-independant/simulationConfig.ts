import { SimulationConfig } from '@/domaine/SimulationConfig'

export const config: SimulationConfig = {
	'objectifs exclusifs': [
		'déclaration charge sociales . résultat . cotisations obligatoires',
		'déclaration charge sociales . résultat . total charges sociales déductible',
		'déclaration charge sociales . résultat . revenu net fiscal',
		'déclaration charge sociales . résultat . assiette sociale',
	],
	situation: {
		'déclaration charge sociales': 'oui',
		'dirigeant . régime social': "'indépendant'",
		date: '01/01/2024',
		'dirigeant . indépendant . PL . CIPAV': 'non',
		'entreprise . catégorie juridique': "''",
		impôt: 'non',
	},
	'unité par défaut': '€',
}
