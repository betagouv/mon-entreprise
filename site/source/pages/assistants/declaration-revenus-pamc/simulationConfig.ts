import { SimulationConfig } from '@/domaine/SimulationConfig'

export const configDéclarationRevenusPAMC: SimulationConfig = {
	situation: {
		'déclaration revenus PAMC': 'oui',
	},
	'unité par défaut': '€',
}
