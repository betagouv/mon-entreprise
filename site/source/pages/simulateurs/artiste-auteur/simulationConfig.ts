import { SimulationConfig } from '../_configs/types'

export const configArtisteAuteur: SimulationConfig = {
	objectifs: [
		'artiste-auteur . cotisations',
		'artiste-auteur . revenus',
		'artiste-auteur . cotisations . IRCEC',
	],
	situation: {
		'artiste-auteur': 'oui',
		dirigeant: {
			'applicable si': 'non',
		},
	},
	'unité par défaut': '€/an',
	questions: {
		liste: [
			'artiste-auteur . revenus . BNC . micro-bnc',
			'artiste-auteur . cotisations . option surcotisation',
			'artiste-auteur . cotisations . IRCEC . cotisation RAAP . taux réduit',
			'artiste-auteur . cotisations . IRCEC . profession',
		],
	},
}
