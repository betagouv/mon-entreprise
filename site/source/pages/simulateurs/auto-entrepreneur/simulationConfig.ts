import { SimulationConfig } from '../_configs/types'

export const configAutoEntrepreneur: SimulationConfig = {
	'objectifs exclusifs': [
		"dirigeant . auto-entrepreneur . chiffre d'affaires",
		'dirigeant . auto-entrepreneur . revenu net',
		'dirigeant . auto-entrepreneur . revenu net . après impôt',
	],
	objectifs: [
		'dirigeant . auto-entrepreneur . cotisations et contributions',
		'dirigeant . rémunération . impôt',
	],
	questions: {
		"à l'affiche": [
			{
				label: "Type d'activité",
				dottedName: 'entreprise . activité . nature',
			},
			{
				label: 'Versement libératoire',
				dottedName:
					'dirigeant . auto-entrepreneur . impôt . versement libératoire',
			},
			{
				label: 'Impôt sur le revenu',
				dottedName: 'impôt . méthode de calcul',
			},
			{
				label: 'ACRE',
				dottedName: 'dirigeant . exonérations . ACRE',
			},
		],
		liste: [
			'entreprise',
			'dirigeant',
			'impôt',
			'établissement',
			'situation personnelle',
		],
		'non prioritaires': ['établissement . commune'],
		'liste noire': [
			'entreprise . activités',
			'entreprise . activités . revenus mixtes',
			'entreprise . charges',
			"entreprise . chiffre d'affaires",
		],
	},
	'unité par défaut': '€/an',
	situation: {
		salarié: 'non',
		'entreprise . catégorie juridique': "'EI'",
		'entreprise . catégorie juridique . EI . auto-entrepreneur': 'oui',
	},
}
