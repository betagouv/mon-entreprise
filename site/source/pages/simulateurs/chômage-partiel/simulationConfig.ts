import { SimulationConfig } from '../_configs/types'

export const configChômagePartiel: SimulationConfig = {
	objectifs: [
		'salarié . contrat . salaire brut',
		'salarié . rémunération . net . à payer avant impôt',
		'salarié . activité partielle . net habituel',
		'salarié . activité partielle . total employeur habituel',
		'salarié . coût total employeur',
	],
	questions: {
		liste: [
			'salarié . activité partielle',
			'salarié . temps de travail',
			'établissement . commune',
		],
	},
	'unité par défaut': '€/mois',
	situation: {
		dirigeant: 'non',
		'salarié . activité partielle': 'oui',
		'salarié . cotisations . prévoyances': {
			'applicable si': 'non',
		},
	},
}
