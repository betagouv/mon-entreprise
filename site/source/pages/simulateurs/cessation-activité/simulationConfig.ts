import { SimulationConfig } from '@/pages/simulateurs/_configs/types'
import { configIndépendant } from '@/pages/simulateurs/indépendant/simulationConfig'

export const configCessationActivité: SimulationConfig = {
	...configIndépendant,
	'unité par défaut': '€/an',
}
