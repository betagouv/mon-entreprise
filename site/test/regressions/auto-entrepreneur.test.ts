import { expect, it } from 'vitest'

import { configAutoEntrepreneur } from '@/pages/simulateurs/auto-entrepreneur/simulationConfig'

import autoEntrepreneurSituations from './auto-entrepreneur.yaml'
import { engine, getMissingVariables, runSimulations } from './utils'

it('calculate simulations-auto-entrepreneur', () => {
	runSimulations(
		autoEntrepreneurSituations,
		[
			...(configAutoEntrepreneur['objectifs exclusifs'] ?? []),
			...(configAutoEntrepreneur.objectifs ?? []),
			'dirigeant . auto-entrepreneur . cotisations et contributions . cotisations',
			'dirigeant . auto-entrepreneur . cotisations et contributions . cotisations . service BIC . taux',
			'dirigeant . auto-entrepreneur . cotisations et contributions . cotisations . service BIC',
			'dirigeant . auto-entrepreneur . cotisations et contributions . cotisations . service BNC . taux',
			'dirigeant . auto-entrepreneur . cotisations et contributions . cotisations . service BNC',
			'dirigeant . auto-entrepreneur . cotisations et contributions . cotisations . vente restauration hébergement . taux',
			'dirigeant . auto-entrepreneur . cotisations et contributions . cotisations . vente restauration hébergement',
			'dirigeant . auto-entrepreneur . cotisations et contributions . cotisations . répartition . maladie-maternité',
			'dirigeant . auto-entrepreneur . cotisations et contributions . cotisations . répartition . invalidité-décès',
			'dirigeant . auto-entrepreneur . cotisations et contributions . cotisations . répartition . retraite de base',
			'dirigeant . auto-entrepreneur . cotisations et contributions . cotisations . répartition . retraite complémentaire',
			'dirigeant . auto-entrepreneur . cotisations et contributions . cotisations . répartition . autres contributions',
		],
		configAutoEntrepreneur.situation
	)

	expect(
		getMissingVariables(
			engine
				.setSituation({
					...configAutoEntrepreneur.situation,
					"dirigeant . auto-entrepreneur . chiffre d'affaires": '30000 €/an',
				})
				.evaluate('dirigeant . auto-entrepreneur . revenu net . après impôt')
		)
	).toMatchInlineSnapshot(`
		[
		  "dirigeant . auto-entrepreneur . impôt . versement libératoire",
		  "entreprise . activité . nature",
		  "entreprise . activités . revenus mixtes",
		  "entreprise . activités . service ou vente",
		  "entreprise . date de création",
		  "impôt . foyer fiscal . enfants à charge",
		  "impôt . foyer fiscal . revenu imposable . autres revenus imposables",
		  "impôt . foyer fiscal . situation de famille",
		  "impôt . méthode de calcul",
		  "établissement . commune . département",
		  "établissement . commune . département . outre-mer",
		]
	`)
})
