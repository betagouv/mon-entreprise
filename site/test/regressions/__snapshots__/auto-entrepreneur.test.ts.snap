// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`calculate simulations-auto-entrepreneur > ACRE 1`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 21345
dirigeant . auto-entrepreneur . cotisations et contributions: 112
dirigeant . auto-entrepreneur . revenu net: 20000
dirigeant . auto-entrepreneur . revenu net . après impôt: 20000
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > ACRE 2`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 32017
dirigeant . auto-entrepreneur . cotisations et contributions: 168
dirigeant . auto-entrepreneur . revenu net: 30000
dirigeant . auto-entrepreneur . revenu net . après impôt: 30000
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > DROM 1`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 20000
dirigeant . auto-entrepreneur . cotisations et contributions: 213
dirigeant . auto-entrepreneur . revenu net: 17440
dirigeant . auto-entrepreneur . revenu net . après impôt: 17440
dirigeant . rémunération . impôt: 0"
`;

exports[`calculate simulations-auto-entrepreneur > DROM 2`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 45000
dirigeant . auto-entrepreneur . cotisations et contributions: 312
dirigeant . auto-entrepreneur . revenu net: 41253
dirigeant . auto-entrepreneur . revenu net . après impôt: 41253
dirigeant . rémunération . impôt: 0"
`;

exports[`calculate simulations-auto-entrepreneur > activité mixte 1`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 20000
dirigeant . auto-entrepreneur . cotisations et contributions: 281
dirigeant . auto-entrepreneur . revenu net: 16630
dirigeant . auto-entrepreneur . revenu net . après impôt: 16630
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > activité mixte 2`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 20000
dirigeant . auto-entrepreneur . cotisations et contributions: 306
dirigeant . auto-entrepreneur . revenu net: 16333
dirigeant . auto-entrepreneur . revenu net . après impôt: 16333
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > activité mixte 3`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 20000
dirigeant . auto-entrepreneur . cotisations et contributions: 290
dirigeant . auto-entrepreneur . revenu net: 16518
dirigeant . auto-entrepreneur . revenu net . après impôt: 16518
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > activité mixte 4`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: undefined
dirigeant . auto-entrepreneur . cotisations et contributions: 689
dirigeant . auto-entrepreneur . revenu net: undefined
dirigeant . auto-entrepreneur . revenu net . après impôt: undefined
dirigeant . rémunération . impôt: 670"
`;

exports[`calculate simulations-auto-entrepreneur > activités 1`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 20000
dirigeant . auto-entrepreneur . cotisations et contributions: 355
dirigeant . auto-entrepreneur . revenu net: 15740
dirigeant . auto-entrepreneur . revenu net . après impôt: 15740
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > activités 2`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 20000
dirigeant . auto-entrepreneur . cotisations et contributions: 357
dirigeant . auto-entrepreneur . revenu net: 15720
dirigeant . auto-entrepreneur . revenu net . après impôt: 15720
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > activités 3`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 20000
dirigeant . auto-entrepreneur . cotisations et contributions: 214
dirigeant . auto-entrepreneur . revenu net: 17432
dirigeant . auto-entrepreneur . revenu net . après impôt: 17432
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > activités 4`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 20000
dirigeant . auto-entrepreneur . cotisations et contributions: 366
dirigeant . auto-entrepreneur . revenu net: 15604
dirigeant . auto-entrepreneur . revenu net . après impôt: 15604
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > activités 5`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 20000
dirigeant . auto-entrepreneur . cotisations et contributions: 207
dirigeant . auto-entrepreneur . revenu net: 17520
dirigeant . auto-entrepreneur . revenu net . après impôt: 17520
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > activités 6`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 20000
dirigeant . auto-entrepreneur . cotisations et contributions: 356
dirigeant . auto-entrepreneur . revenu net: 15728
dirigeant . auto-entrepreneur . revenu net . après impôt: 15728
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > aides 1`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 5708
dirigeant . auto-entrepreneur . cotisations et contributions: 59
dirigeant . auto-entrepreneur . revenu net: 5000
dirigeant . auto-entrepreneur . revenu net . après impôt: 5000
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > aides 2`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 57092
dirigeant . auto-entrepreneur . cotisations et contributions: 591
dirigeant . auto-entrepreneur . revenu net: 50000
dirigeant . auto-entrepreneur . revenu net . après impôt: 50000
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > impôt sur le revenu 1`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 31766
dirigeant . auto-entrepreneur . cotisations et contributions: 564
dirigeant . auto-entrepreneur . revenu net: 25000
dirigeant . auto-entrepreneur . revenu net . après impôt: 24301
dirigeant . rémunération . impôt: 699

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > échelle de revenus 1`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 571
dirigeant . auto-entrepreneur . cotisations et contributions: 6
dirigeant . auto-entrepreneur . revenu net: 500
dirigeant . auto-entrepreneur . revenu net . après impôt: 500
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > échelle de revenus 2`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 1142
dirigeant . auto-entrepreneur . cotisations et contributions: 12
dirigeant . auto-entrepreneur . revenu net: 1000
dirigeant . auto-entrepreneur . revenu net . après impôt: 1000
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > échelle de revenus 3`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 2283
dirigeant . auto-entrepreneur . cotisations et contributions: 24
dirigeant . auto-entrepreneur . revenu net: 2000
dirigeant . auto-entrepreneur . revenu net . après impôt: 2000
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > échelle de revenus 4`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 5708
dirigeant . auto-entrepreneur . cotisations et contributions: 59
dirigeant . auto-entrepreneur . revenu net: 5000
dirigeant . auto-entrepreneur . revenu net . après impôt: 5000
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > échelle de revenus 5`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 11416
dirigeant . auto-entrepreneur . cotisations et contributions: 118
dirigeant . auto-entrepreneur . revenu net: 10000
dirigeant . auto-entrepreneur . revenu net . après impôt: 10000
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > échelle de revenus 6`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 22831
dirigeant . auto-entrepreneur . cotisations et contributions: 236
dirigeant . auto-entrepreneur . revenu net: 20000
dirigeant . auto-entrepreneur . revenu net . après impôt: 20000
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > échelle de revenus 7`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 57092
dirigeant . auto-entrepreneur . cotisations et contributions: 591
dirigeant . auto-entrepreneur . revenu net: 50000
dirigeant . auto-entrepreneur . revenu net . après impôt: 50000
dirigeant . rémunération . impôt: 0

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > échelle de revenus 8`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 79923
dirigeant . auto-entrepreneur . cotisations et contributions: 827
dirigeant . auto-entrepreneur . revenu net: 70000
dirigeant . auto-entrepreneur . revenu net . après impôt: 68974
dirigeant . rémunération . impôt: 1026

Notifications affichées : entreprise . TVA . franchise de TVA . notification"
`;

exports[`calculate simulations-auto-entrepreneur > échelle de revenus 9`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 114169
dirigeant . auto-entrepreneur . cotisations et contributions: 1181
dirigeant . auto-entrepreneur . revenu net: 100000
dirigeant . auto-entrepreneur . revenu net . après impôt: 96781
dirigeant . rémunération . impôt: 3219"
`;

exports[`calculate simulations-auto-entrepreneur > échelle de revenus 10`] = `
"dirigeant . auto-entrepreneur . chiffre d'affaires: 1141745
dirigeant . auto-entrepreneur . cotisations et contributions: 11812
dirigeant . auto-entrepreneur . revenu net: 1000000
dirigeant . auto-entrepreneur . revenu net . après impôt: 871425
dirigeant . rémunération . impôt: 128575

Notifications affichées : entreprise . imposition . régime . micro-entreprise . alerte seuil dépassés"
`;
